src/layout/Button/index.tsx

```diff
@@ -1,20 +1,15 @@
 import React from 'react';
 
-import { ComponentType } from 'src/layout';
 import { ButtonComponent } from 'src/layout/Button/ButtonComponent';
-import { LayoutComponent } from 'src/layout/LayoutComponent';
+import { ActionComponent } from 'src/layout/LayoutComponent';
 import type { PropsFromGenericComponent } from 'src/layout';
 
-export class Button extends LayoutComponent<'Button'> {
+export class Button extends ActionComponent<'Button'> {
   render(props: PropsFromGenericComponent<'Button'>): JSX.Element | null {
     return <ButtonComponent {...props} />;
   }
 
   renderWithLabel(): boolean {
```

Comment:
Not sure what you mean by re-implement when needeed. This functionality is in use, as some components expect `GenericComponent` to render them with a label first, while others do not. The default is then to render with label (returns true), but the Button component should be rendered without, thus is overrides the base implementation and returns false instead.

The philosophy here, I think, is that new components have to implement some functions where there are no sane defaults (making them `abstract` in the base class, so that not implementing them is a compile-time error). For others, such as this one, the sane default is to always render with a label, but if that's not what you wanted you can override the method and provide your own result.

So for Summary, every `FormComponent` now needs to implement a `renderSummary` (and so every developer needs to active think about "how should this component be displayed in a summary?" when implementing a new component type). Previously, most components have not even been tested in a Summary, because there was a default (but not always very functional) implementation - which lived far away from your component implementation.

Not sure if that all made sense, but the TL;DR is that no, we can't remove this thing - it's in use.. 😊

Reply:
Of course, it all makes sense now. Thanks for the clarification.  🚀